letta/agent.py

```diff
@@ -607,6 +604,13 @@ def _handle_ai_response(
                 self.interface.function_message(f"Error: {error_msg}", msg_obj=messages[-1])
                 return messages, False, True  # force a heartbeat to allow agent to handle error
 
+            # Community noted that inner thoughts can sometimes appear in the function args
```

Comment:
And delete this code (if we're using `inner_thoughts` in `function_args`, it should be popped at a different level of the code (inside of the `llm_api.py` code)

Reply:
ah yes, sounds good!